script/pull_production_forecasts.py

```diff
@@ -0,0 +1,101 @@
+"""This scripts fetches solar generation forecasts for Great Britain using the Quartz API.
+
+The API requires authentication via a Bearer token and returns forecast data for a specified time
+period. The forecast data includes predicted solar generation values at half hourly intervals.
+
+Example:
+    - Set access token in the "get_solar_forecast" function at "INSERT ACCESS TOKEN HERE".
+    - Set the start_time to when you want to pull forecasts from.
+    - Adjust the days parameter in end_time variable. (example is a week)
+
+Notes:
+    - Before 2024-06-01 adjustments where made to how pvalues are stored, which this current script
+      does not account for.
+"""
+
+from datetime import datetime, timedelta
+
+import pandas as pd
+import requests
+from tqdm import tqdm
+
+start_time = datetime(2024, 6, 1, 3)
+end_time = start_time + timedelta(days=7)  # Example pull a week of data
+output_file_path = "./quartz_production_backtest.csv"
+
+
+def get_solar_forecast(start_datetime):
+    """
+    Get solar forecast for GB from Quartz API for a specific start time and creation limit
+
+    Args:
+        start_datetime (str): Start datetime in format "YYYY-MM-DD"
+
+    Returns:
+        pandas.DataFrame: Forecast data
+    """
+    # Convert string to datetime object
+    start_dt = datetime.strptime(start_datetime, "%Y-%m-%dT%H%M")
+    end_dt = start_dt + timedelta(days=2)
+
+    # Convert back to string format for API
+    end_datetime = end_dt.strftime("%Y-%m-%dT%H%M")
+    creation_limit = start_datetime
+
+    # API endpoint and parameters
+    url = "https://api.quartz.solar/v0/solar/GB/national/forecast"
+    params = {
+        "include_metadata": "false",
+        "start_datetime_utc": start_datetime,
```

Comment:
you can might want to use toisoformat, https://www.geeksforgeeks.org/isoformat-method-of-datetime-class-in-python/, but maybe its ok, just leaving it as a datetime object